src/power_grid_model/core/power_grid_model.py

```diff
@@ -349,14 +349,14 @@ def _calculate_state_estimation(
         symmetric: bool = True,
         error_tolerance: float = 1e-8,
         max_iterations: int = 20,
-        calculation_method: Union[CalculationMethod, str] = CalculationMethod.iterative_linear,
+        calculation_method: CalculationMethod | str = CalculationMethod.iterative_linear,
         update_data: Optional[Dataset] = None,
         threading: int = -1,
-        output_component_types: Optional[Union[Set[ComponentType], List[ComponentType]]] = None,
+        output_component_types: Optional[Set[ComponentType] | list[ComponentType]] = None,
```

Comment:
please double-check that all cases of List, Tuple, Set, Dict have been removed
```suggestion
        output_component_types: Optional[set[ComponentType] | list[ComponentType]] = None,
```